src/main/java/twilightforest/entity/boss/EntityTFNaga.java

```diff
@@ -650,16 +659,13 @@ private boolean isEntityWithinHomeArea(Entity entity)
 
 	private void spawnBodySegments()
 	{
-		if (!world.isRemote)
+		for (int i = 0; i < currentSegmentCount; i++)
 		{
-			for (int i = 0; i < currentSegmentCount; i++)
+			if (body[i] == null || body[i].isDead)
 			{
-				if (body[i] == null || body[i].isDead)
-				{
-					body[i] = new EntityTFNagaSegment(this, i);
-					body[i].setLocationAndAngles(posX + 0.1 * i, posY + 0.5D, posZ + 0.1 * i, rand.nextFloat() * 360F, 0.0F);
-					world.spawnEntity(body[i]);
-				}
+				body[i] = new EntityTFNagaSegment(this, i);
+				body[i].setLocationAndAngles(posX + 0.1 * i, posY + 0.5D, posZ + 0.1 * i, rand.nextFloat() * 360F, 0.0F);
+				if (!world.isRemote) world.spawnEntity(body[i]); // Moved the !world.isRemote check here as the segments were not spawning on the client side otherwise for some reason.
```

Comment:
About this one, I wasn't exactly sure and spent a good while looking into it before, what i'm thinking is that the segments required data of the naga of somesort on the client side.. but tests made that untrue, and the code itself still doesnt quite make sense why this would work over the previous. I really coulnd't figure it out.

My other idea is that the naga itself on the client side needed its body array filled else it wouldnt show the segments, this one I suppose makes a bit more sense but still not sure to why exactly.